### ISSUE TITLE
BUILD.py - TypeError with option 'wb'

### ISSUE THREAD

**Issue by speedchase** [brandon-rhodes/pycon-pandas-tutorial#18]
On Windows with Python 3.5, I get the following error with  'wb' option for the open file statement. As advised, with just the 'w' option, blank lines are added to the file. Below is the error message.

\build\BUILD.py", line 58, in main
    output.writerow(('title', 'year'))
TypeError: a bytes-like object is required, not 'str'

**Comment by speedchase** [comment]
the issue has been resolved by using the newline='' parameter for csv open
`with open('../data/titles.csv', 'w', newline='') as f:`

**Comment by alexsandracodes** [comment]
I added that line, "with open(......) as f:", and it worked for the first part which was writing titles.csv. However, the problem persists with the other outputs, like release-dates. I added this "with open" line to the subsequent output writing lines, but the same error occurs:

```
Reading release dates from "release-dates.list.gz"
Traceback (most recent call last):
  File "build/BUILD.py", line 226, in <module>
  File "build/BUILD.py", line 100, in main
ValueError: I/O operation on closed file.
```

**Comment by brandon-rhodes** [comment]
Interesting! So, simply changing the `wb` to `w` did not fix it? Did it at least change the error message, or did the error message stay the same? What traceback do you get if you try `wb` with `newline=''` or even `newline=b''`?

**Comment by mrestko** [comment]
Just had this problem too and found that speedchase's solution works. Trying `newline=''` with mode `'wb'` gives an error complaining that `newline` can't be used with binary mode.

For what it's worth, I'm running this script on Windows Subsystem for Linux.

**Comment by brandon-rhodes** [comment]
Alas — as this issue dates from back before several subsequent pull requests addressing Windows and how files are opened, I don't have any way to determine whether folks are still hitting this problem. I'm going to close, but please re-open if the subsequent changes haven't addressed this particular Python 3.5 problem!
